gnucash/gnome-utils/gnc-amount-edit.c

```diff
@@ -257,19 +257,67 @@ gnc_amount_edit_new (void)
     return GTK_WIDGET(gae);
 }
 
+static gint
+get_original_error_position (const gchar *string, const gchar *symbol,
+                             gint error_pos)
+{
+    gchar **split_text;
+    gint split_text_len;
+    gint temp = 0;
+    gint ret = 0;
+    gint x = 0;
+    gboolean found = FALSE;
+
+    if (error_pos == 0)
+        return ret;
+
+    split_text = g_strsplit (string, symbol, -1);
+    split_text_len = g_strv_length (split_text);
+
+    for (x; x < split_text_len; x++)
+    {
+         gint s_len = g_utf8_strlen (split_text[x], -1);
+
+         temp = temp + s_len;
+
+         if (error_pos <= temp)
+         {
+             found = TRUE;
+             break;
+         }
+    }
+
+    if (found)
+       ret = error_pos + (x * g_utf8_strlen (symbol, -1));
+    else
+       ret = g_utf8_strlen (string, -1) - 1;
```

Comment:
How can `error_len` be larger than the sum of the lengths of `split_text`'s pieces?

Reply:
I need to do some testing, I added this for a reason when I started but maybe I have changed a part and not checked if is needed now.